cbor.js

```diff
@@ -23,9 +23,63 @@
  */
 
 (function(global, undefined) { "use strict";
-var POW_2_24 = 5.960464477539063e-8,
-    POW_2_32 = 4294967296,
-    POW_2_53 = 9007199254740992;
+var POW_2_32 = 4294967296,
+    POW_2_53 = 9007199254740992,
+    singleFloatView = new Float32Array(1),
+    singleIntView = new Uint32Array(singleFloatView.buffer);
+
+
+function getFloat16(value) {
+  singleFloatView[0] = value;
+  var fbits = singleIntView[0];
+  var sign = ( fbits >> 16 ) & 0x8000;
+  var exponentAndMantissa = fbits & 0x7fffffff;
+  var val = ( exponentAndMantissa ) + 0x1000;
+  if (val >= 0x47800000) {
+    /* jshint laxbreak:true */
+    return val < 0x7f800000
+      ? sign | 0x7c00
+      : sign | 0x7c00 | ( fbits & 0x007fffff ) >> 13;
+    /* jshint laxbreak:false */
+  }
+  if (val >= 0x38800000) {
+    return sign | val - 0x38000000 >> 13;
+  }
+  if (val < 0x33000000) {
+    return sign;
+  }
+  val = ( exponentAndMantissa ) >> 23;
+  return sign | ( ( fbits & 0x7fffff | 0x800000 ) +
+    ( 0x800000 >>> val - 102 )
+    >> 126 - val );
+}
+
+function decodeFloat16(value) {
+  var exponent = (value & 0x7C00) >> 10,
+    fraction = value & 0x03FF;
+  return (value >> 15 ? -1 : 1) * (
+      exponent ?
+        (
+          exponent === 0x1F ?
+            fraction ? NaN : Infinity :
+          Math.pow(2, exponent - 15) * (1 + fraction / 0x400)
+        ) :
+      6.103515625e-5 * (fraction / 0x400)
+    );
+}
+
+function checkFloat16(value) {
+  var data = getFloat16(value);
+  var decoded = decodeFloat16(data);
+  return decoded === value || isNaN(decoded) && isNaN(value) ? data : false;
+}
+
+function checkFloat32(value) {
+  singleFloatView[0] = value;
+  var xf = singleFloatView[0];
+  // skip NaN check, should be encoded as float 16
+  return xf === value ? singleIntView[0] : false;
+}
```

Comment:
do you think it's a lot more work to do a clean check if Float64 will fit in a smaller type? If would be ok if we would only check for one other type, but checking first for Float16 and then for Float32 and finally writing a Float64 does not seam like an good solution for me. (The be honest: Not having a good check was the main reason for skipping the Float16/Float32 enconding)